packages/core/src/fetchers/fetchAllEntities.ts

```diff
@@ -1,66 +1,125 @@
-import { ContentfulClientApi, Entry } from 'contentful';
+import { Asset, ContentfulClientApi, Entry } from 'contentful';
 import { MinimalEntryCollection } from './gatherAutoFetchedReferentsFromIncludes';
 
 const MIN_FETCH_LIMIT = 1;
-
-const fetchEntities = async ({
-  entityType,
+export const fetchAllEntries = async ({
   client,
-  query,
+  ids,
+  locale,
+  skip = 0,
+  limit = 100,
+  responseItems = [],
+  responseIncludes = { Entry: [], Asset: [] },
 }: {
-  entityType: 'Entry' | 'Asset';
   client: ContentfulClientApi<undefined>;
-  query: { 'sys.id[in]': string[]; locale: string; limit: number; skip: number };
+  ids: string[];
+  locale: string;
+  skip?: number;
+  limit?: number;
+  responseItems?: Entry[];
+  responseIncludes?: MinimalEntryCollection['includes'];
 }) => {
-  if (entityType === 'Asset') {
-    return client.getAssets({ ...query });
-  }
+  try {
+    if (!client) {
+      throw new Error(
+        'Failed to fetch experience entities. Required "client" parameter was not provided',
+      );
+    }
+
+    if (!ids.length) {
+      return {
+        items: [],
+        includes: {
+          Entry: [],
+          Asset: [],
+        },
+      };
+    }
+
+    const query = { 'sys.id[in]': ids, locale, limit, skip };
+
+    const {
+      items,
+      includes,
+      total: responseTotal,
+    } = await client.withoutLinkResolution.getEntries({ ...query });
+
+    responseItems.push(...(items as Entry[]));
+    responseIncludes?.Entry?.push(...(includes?.Entry || []));
+    responseIncludes?.Asset?.push(...(includes?.Asset || []));
```

Comment:
subjective: nit: 
From what I understand if `responseIncludes` is available, it _must_ already have `.Entry` and `.Asset` by contract. So secodn `?` is not necessary? 

eg. 
```
responseIncludes?.Entry.push(...(includes?.Entry || []))
```

?